lib/components/user/monitored-trip/trip-basics-pane.js

```diff
@@ -104,22 +97,26 @@ class TripBasicsPane extends Component {
     // but don't show it when they click submit or cancel.
     const unsavedChanges = dirty && !isSubmitting && !canceled
     // Message changes depending on if the new or existing trip is being edited
-    const unsavedChangesNewTripMessage = 'You haven\'t saved your new trip yet. If you leave, it will be lost.'
-    const unsavedChangesExistingTripMessage = 'You haven\'t saved your trip yet. If you leave, changes will be lost.'
-    const unsavedChangesMessage = isCreating ? unsavedChangesNewTripMessage : unsavedChangesExistingTripMessage
+    const unsavedChangesMessage = intl.formatMessage({
+      id: `components.TripBasicsPane.unsavedChanges${isCreating ? 'New' : 'Existing'}Trip`
```

Comment:
Please do not create dynamic message IDs as it is hard to automatically keep track of which IDs are in the code. Instead, please create a separate variable for each possible message.